### PR TITLE
Remove fastcomp-only SEPARATE_ASM option. See #11860

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -3031,9 +2995,6 @@ def consume_arg():
       options.valid_abspaths.append(newargs[i + 1])
       newargs[i] = ''
       newargs[i + 1] = ''
-    elif newargs[i] == '--separate-asm':
```
Comment (sbc100) [inline emcc.py:3034]: Keep this around and issue an appropriate error?

File: src/settings.js [hunk 2]
```diff
@@ -1080,10 +1070,6 @@ var ASM_JS = 1;
 // [fastcomp-only]
 var FINALIZE_ASM_JS = 1;
 
-// see emcc --separate-asm
-// [fastcomp-only]
-var SEPARATE_ASM = 0;
```
Comment (sbc100) [inline src/settings.js:1085]: Add these 2 to LEGACY_SETTINGS.

File: emcc.py (hunk 1 above)
Comment (kripken) [reply emcc.py:3034]: Ah, good point, this PR needs to do that to be NFC, as we errored before. Fixing...

File: src/settings.js (hunk 2 above)
Comment (kripken) [reply src/settings.js:1085]: Done.

**Comment by kripken** [comment]
> Is there no use for this feature with wasm2js?

No use. It's always errored with the wasm backend (and fastcomp with wasm). It's only ever been asm.js specific.

(I can in theory see a need for such a feature for wasm2js though. If that comes up we can revive this, or maybe find a better way.)

**Pull request merged**